app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
-   * @param newUser - A modified user object intended to be saved.
-   * @param oldUser - The original user object of `newUser`. Returned in case API call fails
-   *
-   */
   sendUserToServer(newUser: APIUserType, oldUser: APIUserType): Promise<APIUserType> {
     return updateUser(newUser).then(() => Promise.resolve(newUser), () => Promise.reject(oldUser));
   }
 
-  closeModal(usersPromises: Array<Promise<APIUserType>>): void {
+  resolvePromisesAndCloseModal(usersPromises: Array<Promise<APIUserType>>): void {
     Promise.all(usersPromises).then(
       newUsers => {
         this.setState({
-          domain: null,
-          level: null,
+          multipleUsersEntries: [],
+          singleUsersEntries: [],
         });
         this.props.onChange(newUsers);
       },
-      () => {
-        // do nothing and keep modal open
+      error => {
+        handleGenericError(error);
       },
     );
   }
 
-  render() {
-    if (!this.props.visible) {
-      return null;
+  recordModifiedAndExistedBefore = (record: SingleUserTableEntry): boolean => {
+    if (this.props.selectedUsers.length === 1)
+      return (
+        record.value !== this.props.selectedUsers[0].experiences[record.domain] &&
+        record.domain in this.props.selectedUsers[0].experiences
+      );
+    else return false;
+  };
+
+  setValueOfEntry = (index: number, value: number) => {
+    if (value > 0) {
+      if (this.props.selectedUsers.length === 1) {
+        this.setState(prevState => ({
+          singleUsersEntries: prevState.singleUsersEntries.map((entry, currentIndex) => {
+            if (currentIndex === index) {
+              return {
+                ...entry,
+                value,
+              };
+            } else {
+              return entry;
+            }
+          }),
+        }));
+      } else {
+        this.setState(prevState => ({
+          multipleUsersEntries: prevState.multipleUsersEntries.map((entry, currentIndex) => {
+            if (currentIndex === index) {
+              return {
+                ...entry,
+                value,
+              };
+            } else {
+              return entry;
+            }
+          }),
+        }));
+      }
+    }
+  };
+
+  revertChangesOfEntry = (index: number) => {
+    if (this.props.selectedUsers.length === 1) {
+      this.setState(prevState => ({
+        singleUsersEntries: prevState.singleUsersEntries.map((entry, currentIndex) => {
+          if (currentIndex === index) {
+            return {
+              ...entry,
+              value: this.props.selectedUsers[0].experiences[entry.domain],
+            };
+          } else {
+            return entry;
+          }
+        }),
+      }));
+    } else {
+      this.setState(prevState => ({
+        multipleUsersEntries: prevState.multipleUsersEntries.map((entry, currentIndex) => {
+          if (currentIndex === index) {
+            return {
+              ...entry,
+              value: entry.lowestValue,
```

Comment:
Why is the value reset to `lowestValue`? I think, this is an overall question I have:

If I select two users which have `sampleExp: 1` and `sampleExp: 3` and want to add another domain, while leaving `sampleExp` untouched; this won't be possible, since the UI defaults to `lowestValue` and thus sets `sampleExp` to `1` for both users. Shouldn't the input field be blank to indicate that the varying range is not overwritten?